metaci/testresults/models.py

```diff
@@ -116,6 +113,18 @@ class TestResult(models.Model):
     method = models.ForeignKey(
         TestMethod, related_name="test_results", on_delete=models.CASCADE
     )
+    # The robot_task field is used to reconstitute the test log,
+    # because the task has a list of options used by robot when the
+    # test was run (e.g. noncritical, tagstatlink, etc)
+    robot_task = models.ForeignKey(
+        "build.FlowTask",
+        related_name="test_results",
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        default=None,
+        help_text="Task which generated the original output.xml file",
+    )
```

Comment:
The TestResult model is for all test results, not just robot tests. Can we name this field `task` instead of `robot_task` to leave open the possibility of linking non-robot test results to their task as well? (It's fine if we leave actually doing that for later; I just don't see a reason to name this in a way that limits it to being for robot tests.)

Reply:
I'm glad you brought that up. It's tempting to name it `task`, but the code only sets it for robot tasks. I don't think there's any way to know how to set it for other records. The way I set it now is I look at the timestamp of the output.xml file and then find a task that started and ended around that time. Without an output.xml file I don't know how to set it. If there's a better way for a specific task to associate itself with some test results, I'm all ears. 

I have no problem renaming it `task` if you think that makes more sense. I can just add a comment that says says it's currently only valid for robot tasks.

Reply:
I was tempted to create a separate model specifically for robot tasks and add a foreign key to it in the TestResult mode -- we could use it to add a lot of robot-specific statistics. That seemed like too big of a change at this time, especially for my first foray into the metaci code base, and especially because I don't have my head wrapped around the whole model.

Reply:
I agree adding a separate model for robot test results feels like a big change to tackle right now.

I like the idea of calling it `task` and adding a comment that it's currently only set for robot tests.

Reply:
I renamed the field to `task` and added a brief comment